File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(string[] keymap, string[] targets) {
        int[] answer = new int[targets.Length];
        
        var key = new Dictionary<char, int>();
        
        foreach(string s in keymap)
        {
            for(int i = 0; i < s.Length; i++)
            {
                char ch = s[i];
                if(key.ContainsKey(ch))
                {
                    key[ch] = Math.Min(key[ch], i + 1);
                }
                else
                {
                    key[ch] = i + 1;
                }
            }
        }
            
            for(int i = 0; i < targets.Length; i++)
            {
                for(int j = 0; j < targets[i].Length; j++)
                {
                    char ch = targets[i][j];
                    if(!key.ContainsKey(ch))
                    {
                        answer[i] = -1;
                        break;
                    }
                    else
                    {
                        answer[i] += key[ch];
                    }
                }
            }
        return answer;
    }
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<string> keymap, vector<string> targets) {
    vector<int> answer;
    map<char, int> mp;
    
    for(string key : keymap)
    {
        for(int i = 0; i < key.size(); i++)
        {
            char c = key[i];
            int cnt = i + 1;
            
            if(mp.find(c) == mp.end())
            {
                mp[c] = cnt;
            }
            else
            {
                mp[c] = min(mp[c], cnt);
            }
        }
    }
    
    for(string target : targets)
    {
        int sum = 0;
        bool flag = true;
        
        for(char c : target)
        {
            if(mp.find(c) == mp.end()) {
                flag = false;
                break;
            }
            sum += mp[c];
        }
        
        if(flag) answer.push_back(sum);
        else answer.push_back(-1);
    }
    return answer;
}